### PR TITLE
Add abnormal and increased mass density patterns

### LINKED ISSUE #825: Add new abnormal and increased mass density patterns
1. [x] add new pattern `abnormalMassDensityOfAnatomicalEntity.yaml` for review
* Examples:
    1. HP:0004348	[Abnormality of bone mineral density](https://www.ebi.ac.uk/ols/ontologies/hp/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FHP_0004348)
    2.  HP:0009132	[Abnormal tarsal bone mineral density](https://www.ebi.ac.uk/ols/ontologies/hp/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FHP_0009132)
    3.  MP:0010119	[abnormal bone mineral density](https://www.ebi.ac.uk/ols/ontologies/mp/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FMP_0010119)
2. [x] add new pattern `abnormalMassDensityOfAnatomicalEntityInLocation.yaml` for review
* Examples: ?
3. [x] add new pattern `abnormallyIncreasedMassDensityOfAnatomicalEntity.yaml` for review
* Examples:
    1. HP:0000250	[Dense calvaria](https://www.ebi.ac.uk/ols/ontologies/hp/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FHP_0000250)
    2. HP:0011001	[Increased bone mineral density](https://www.ebi.ac.uk/ols/ontologies/hp/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FHP_0011001)
    3. MP:0000062	[increased bone mineral density](https://www.ebi.ac.uk/ols/ontologies/mp/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FMP_0000062)
4. [x] add new pattern `abnormallyIncreasedMassDensityOfAnatomicalEntityInLocation.yaml` for review
* Example:
    1. HP:0006440	[Increased density of long bone diaphyses](https://www.ebi.ac.uk/ols/ontologies/hp/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FHP_0006440)
5. [x] get feedback
6. [x] fix or improve pattern based on feedback

### Notes
1. [mass density PATO:0001019](https://www.ebi.ac.uk/ols/ontologies/pato/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0001019&viewMode=PreferredRoots&siblings=false)
2. [increased mass density PATO:0001788](https://www.ebi.ac.uk/ols/ontologies/pato/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0001788)

### PR DESCRIPTION
**Pull request by rays22**
If applied, this commit will fix #825.

### CODE REVIEW

File: src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+pattern_name: abnormalMassDensityOfAnatomicalEntity
+
+pattern_iri: http://purl.obolibrary.org/obo/upheno/patterns-dev/abnormalMassDensityOfAnatomicalEntity.yaml
+
+description: 'Abnormal mass density (mass per unit volume) of an anatomical entity, usually a bone, for example abnormal bone mineral density.'
```
Comment (rays22) [inline src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml:6]: Take out 'bone'

File: src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+def:
+  text: "Abnormal mass density (mass per unit volume) of %s."
+  vars:
+    - anatomical_entity
+
+equivalentTo:
+  text: "'has_part' some (
+            'mass_density' and
+            'characteristic_of' some %s and
```
Comment (rays22) [inline src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml:51]: Use more brackets.

```
 ('characteristic_of' some %s)
```

File: src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  has_part: BFO:0000051
+
+annotationProperties:
+  exact_synonym: oio:hasExactSynonym
+
+vars:
+  anatomical_entity: "'anatomical_entity'"
+
+name:
+  text: "abnormal %s density"
```
Comment (rays22) [inline src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml:33]: abnormal %s mass density

File: src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml (hunk 1 above)
Comment (rays22) [reply src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml:6]: Add warning: Not to be used for count per area  (e.g. cell count).

**Comment by sbello** [comment]
okay with me after changes suggest in editors call are made orcid.org/0000-0003-4606-0597

File: src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  vars:
+    - anatomical_entity
+
+annotations:
+  - annotationProperty: exact_synonym
+    text: "abnormal density of %s"
```
Comment (rays22) [inline src/patterns/dosdp-dev/abnormalMassDensityOfAnatomicalEntity.yaml:39]: Change it to 'abnormal **mass** density'.

**Comment by anna-anagnostop** [comment]
ok by me Anna Anagnostopoulos https://orcid.org/0000-0002-6490-7723

**Comment by rays22** [comment]
Related ticket: https://github.com/pato-ontology/pato/issues/481

**Pull request merged**